src/main/java/blockchain/LightChainNode.java

```diff
@@ -629,7 +634,8 @@ public SignedBytes PoV(Transaction t) throws RemoteException {
 			long startTime = System.currentTimeMillis();
 			updateView();
 			isAuth = isAuthenticated(t);
-			isCorrect = isCorrect(t);
+			if (params.getChain() == false) { isCorrect = isCorrect(t);	}
+			else{	isCorrect = isCorrect(t,c);	}
 			isSound = isSound(t);
```

Comment:
this if-else should happen outside of `LightChainNode` class at the time of its construction. See my comment below.